libpod/image/search.go

```diff
@@ -0,0 +1,279 @@
+package image
+
+import (
+	"context"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/containers/image/docker"
+	"github.com/containers/image/types"
+	"github.com/containers/libpod/libpod/common"
+	sysreg "github.com/containers/libpod/pkg/registries"
+	"github.com/fatih/camelcase"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/semaphore"
+)
+
+const (
+	descriptionTruncLength = 44
+	maxQueries             = 25
+	maxParallelSearches    = int64(6)
+)
+
+// SearchResult is holding image-search related data.
+type SearchResult struct {
+	// Index is the image index (e.g., "docker.io" or "quay.io")
+	Index string
+	// Name is the canoncical name of the image (e.g., "docker.io/library/alpine").
+	Name string
+	// Description of the image.
+	Description string
+	// Stars is the number of stars of the image.
+	Stars int
+	// Official indicates if it's an official image.
+	Official string
+	// Automated indicates if the image was created by an automated build.
+	Automated string
+}
+
+// SearchOptions are used to control the behaviour of SearchImages.
+type SearchOptions struct {
+	// Filter allows to filter the results.
+	Filter []string
```

Comment:
Can we nuke this in favor of just making searchFilterParams public and embedding it?

Reply:
What I like about the current version is that it avoids boilerplate code at the call sites, mainly to transform the `[]string` into the `searchFilterParams`. But if you prefer the other way, I can change that. Just let me know what you think after this comment.

Reply:
Can we keep the parsing code common (maybe stick it in cmd/podman/shared?) so we can use the nicer searchFilterParams interface in libpod/image? It seems like a much nicer API for folks trying to use search

Reply:
I think it's worth applying a similar change to the Varlink API changing the `[]string` to a `SearchFilter` (including a change from `*bool` to an opt bool).

Reply:
Done